src/common/CExpression.cpp

```diff
@@ -1016,17 +1036,34 @@ llong CExpression::GetValMath( llong llVal, lpctstr & pExpr )
 			++pExpr;
 			{
 				llong iVal = GetVal( pExpr );
+				bool isNegative = false;

```

Comment:
For this, the way I did it same as explained here but if power command do it automatically I can change it to default. I will make some tests then pull it :)

Reply:
Power doesn't do that automatically sadly, because if one of those particular parameters are passed it will instead change the global errno value, you don't know if the operation was successiful just from the return value. It's better to do the checks before calling power

Reply:
I checked and see power work fine with negative values, I updated file and pulled again with a new fix, that only cancel negative value error on Sphere.

Reply:
Would you like to add the checks on power operands here on this pull request, or do you prefer me to do that on a separate commit?

Reply:
What type of checks? Because I think the only check for error is already there, for negative values seems like pow working fine. Am I missing something?

Reply:
Oh good, i didn't realize that snprintf in CFloatMath::FloatMath handled those special values and i thought we needed manual error checking. Works good then!